packages/overlays/src/OverlayController.js

```diff
@@ -891,97 +894,90 @@ export class OverlayController extends EventTargetShim {
+   * it is removed. Otherwise this is the first time displaying a backdrop, so a animation-in
    * animation is played.
    * @param {{ animation?: boolean, phase: OverlayPhase }} config
    */
-  _handleBackdrop({ animation = true, phase }) {
-    if (this.placementMode === 'local') {
-      switch (phase) {
-        case 'init':
-          if (!this.backdropNode) {
-            this.__backdropNode = document.createElement('div');
-            /** @type {HTMLElement} */
-            (this.backdropNode).classList.add('local-overlays__backdrop');
-          }
-          this.backdropNode.slot = '_overlay-shadow-outlet';
-          /** @type {HTMLElement} */
-          (this.contentNode.parentNode).insertBefore(this.backdropNode, this.contentNode);
-          break;
-        case 'show':
-          this.__hasActiveBackdrop = true;
-          break;
-        case 'hide':
-          if (!this.backdropNode) {
-            return;
-          }
-          this.__hasActiveBackdrop = false;
-          break;
-        case 'teardown':
-          if (!this.backdropNode || !this.backdropNode.parentNode) {
-            return;
-          }
-          this.backdropNode.parentNode.removeChild(this.backdropNode);
-          this.__backdropNode = undefined;
-          break;
-        /* no default */
-      }
-      return;
-    }
+  _handleBackdrop({ phase }) {
     switch (phase) {
-      case 'init':
-        this.__backdropNode = document.createElement('div');
-        this.backdropNode.classList.add('global-overlays__backdrop');
-        /** @type {HTMLElement} */
-        (this.contentWrapperNode.parentElement).insertBefore(
-          this.backdropNode,
-          this.contentWrapperNode,
-        );
+      case 'init': {
+        if (!this.backdropNode) {
+          this.__backdropNode = document.createElement('div');
+          /** @type {HTMLElement} */
+          (this.backdropNode).classList.add(`${this.placementMode}-overlays__backdrop`);
+        }
+        this.backdropNode.slot = 'backdrop';
+
+        let insertionAnchor = /** @type {HTMLElement} */ (this.contentNode.parentNode);
+        let insertionBefore = this.contentNode;
+        if (this.placementMode === 'global') {
+          insertionAnchor = /** @type {HTMLElement} */ (this.contentWrapperNode.parentElement);
+          insertionBefore = this.contentWrapperNode;
+        }
+        insertionAnchor.insertBefore(this.backdropNode, insertionBefore);
         break;
+      }
       case 'show':
-        this.backdropNode.classList.add('global-overlays__backdrop--visible');
-        if (animation === true) {
-          this.backdropNode.classList.add('global-overlays__backdrop--fade-in');
+        this.backdropNode.classList.add(`${this.placementMode}-overlays__backdrop--visible`);
+        if (this.config?.hasBackdropAnimation === true) {
+          this.backdropNode.classList.add(`${this.placementMode}-overlays__backdrop--animation-in`);
```

Comment:
maybe we can do this as a default implimentation ofn transitionShow? So the user can override it?
Then this would be the only place where we have a default animation (kind of fine, since native <dialog> has it as well I think)
And all other apis would be the same:
Animations can be done via Promise functions (for maximum flexibility (you can use greensock whatever) and are not toggled via boolean flags)